### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by sameera2004**

### CODE REVIEW

File: skxray/roi.py [hunk 1]
```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
     edges = np.cumsum(steps).reshape(-1, 2)
 
     return edges
+
+
+def divide_pies(image_shape, radius, calibrated_center,
```
Comment (danielballan) [inline skxray/roi.py]: `pie_slices(edges, slicing, center, shape, theta0=0)` where `slicing` can be an integer (`slicing=8` gives 8 equal-sized slices, like a pizza) or a list of angles in degrees like `[0, 70, 190]`. In either case `theta0` should be added to the resultant angles to provide an easy way to adjust the alignment.

File: skxray/roi.py [hunk 2]
```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
...
+    num_angles : int
+        number of angles ring divide into
+        angles are measured from horizontal-anti clock wise
+
+    rotate : {'Y', 'N'}, optional
+        to make angles measured from vertical-anti clock wise
+
+    Returns
+    -------
+    labels_grid : array
```
Comment (danielballan) [inline skxray/roi.py]: Call this `label_array` for consistency.

File: skxray/roi.py [hunk 3]
```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
...
+    calibrated_center : tuple
+        defining the center of the image
+        (column value, row value) (mm)
+
+    Returns
+    -------
+    angle_grid : array
+        angle values from the calibrated center
+        shape image_shape
+    """
```
Comment (danielballan) [inline skxray/roi.py:317]: @sameera2004 and I benchmarked this against trackpy's theta_mask (https://github.com/soft-matter/trackpy/blob/master/trackpy/masks.py#L40) and found theta_mask to be 30X times faster. Note that theta_mask assumes r=0 is in the center of the image, which may be the source of perf gains.

File: skxray/roi.py [hunk 4]
```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
...
+
+    Returns
+    -------
+    labels_grid : array
+        indices of the required roi's
+        shape is ([image_shape[0], image_shape[1]])
+    """
+    angle_grid  = get_angle_grid(image_shape, calibrated_center)
+    # required angles
+    angles = np.linspace(0, 360, num_angles)
```
Comment (danielballan) [inline skxray/roi.py]: `np.isscalar` can be used to check whether `angles` needs to be constructed as it is here, or whether we can just use user input.

File: skxray/roi.py [hunk 5]
```diff
@@ -229,3 +231,87 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
...
+
+    calibrated_center : tuple
+        defining the center of the image
+        (column value, row value) (mm)
+
+    num_angles : int
+        number of angles ring divide into
+        angles are measured from horizontal-anti clock wise
+
+    rotate : {'Y', 'N'}, optional
```
Comment (danielballan) [inline skxray/roi.py]: Remember to remove this.

File: skxray/roi.py [hunk 6]
```diff
@@ -229,3 +231,96 @@ def ring_edges(inner_radius, width, spacing=0, num_rings=None):
     edges = np.cumsum(steps).reshape(-1, 2)
 
     return edges
+
+
+def pie_slices(edges, slicing, center, shape, theta=0):
```
Comment (sameera2004) [inline skxray/roi.py:236]: @danielballan I added a pie_slices function and it draw the mask we required. I will add some tests to it

**Comment by danielballan** [comment]
Great!

**Pull request merged**